test/validator/breaking_with_deprecated_test.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test/test.dart';
+
+import '../descriptor.dart' as d;
+import '../test_pub.dart';
+import 'utils.dart';
+
+void main() {
+  test('should only warn when publishing a breaking release ', () async {
+    final server = await servePackages();
+    await d.dir(appPath, [
+      d.validPubspec(extras: {'version': '2.0.0'}),
+      d.file('LICENSE', 'Eh, do what you want.'),
+      d.file('README.md', "This package isn't real."),
+      d.file('CHANGELOG.md', '# 2.0.0\nFirst version\n'),
+      d.dir('lib', [
+        d.file(
+          'test_pkg.dart',
+          "@Deprecated('Stop using this please') int i = 1;",
```

Comment:
Why should this be a problem?

Yes, in theory a major version bump is a good opportunity to remove deprecated stuff.

But it's also entirely possible that my package contains multiple kinds of deprecated stuff, what if I want to keep sonme of the deprecated stuffs and remove other deprecated stuff.

In general, unless there is a pressing reason to remove something that's been deprecated, I would suggest not doing so.
You could do things like `/// @nodoc` in dartdoc documentation comments, to exclude the deprecated bits from documentation, so it doesn't pollute.

Would be nice if there was also a way to hide it from auto-completion.


---------------------------

Yes, in many cases it makes sense to cleanup when doing a major version bump, such that deprecated stuff disappears. But for large complex packages it's entirely possible that some deprecated stuff sticks around because it's low cost to keep, and removing it only causes friction.

Reply:
Yeah - good points...

Maybe we can lower this to  a "hint"... I still think there is some value to extract here (the associated issue seems quite popular)...

Reply:
WDYT @jonasfj ? Should we make this a hint or close the issue as wontfix?

Reply:
I don't mind shipping it as a hint.

Are we sure this shouldn't just be a "lint" or diagnostic in the analyzer?
We run an analyzes run before publishing, right?

Reply:
Hmm - maybe that would work...

@pq what do you say? Would we like to be linted not to have deprecated members if the version is `x.0.0` ?